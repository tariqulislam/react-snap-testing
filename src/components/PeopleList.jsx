import React, {Component} from 'react'
import PropTypes from 'prop-types'


class PeopleList extends Component {
  render() {
    const {lstPeoples} = this.props

    if(!lstPeoples) {
      return <span> Invalide Data Format for People List</span>
    }

    if(!lstPeoples.length) {
      return <span> Empty List View </span>
    }

    return (
        <div>
          {/* Create the table header which is Similar to JSON object
            in /src/DataSource.json file
          */}
            <table>
              <thead>
                 <tr>
                  <th>Id</th>
                  <th>User Name</th>
                  <th>Email</th>
                  <th>Phone</th>
                </tr>
              </thead>
              <tbody>
              {
                lstPeoples && lstPeoples.map((item, index) => {
                    return (
                      <tr key={index}>
                        <td>{item.id}</td>
                        <td>{item.username}</td>
                        <td>{item.email}</td>
                        <td>{item.phone}</td>
                      </tr>
                    )
                })
              }
              </tbody>
            </table>
        </div>
    )
  }
}

PeopleList.propTypes = {
  lstPeoples: PropTypes.array
}

export default PeopleList