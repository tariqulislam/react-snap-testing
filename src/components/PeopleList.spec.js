import React from 'react'
import renderer from 'react-test-renderer'
import PeopleList from './PeopleList'
import lstPeoples from '../DataSource.json'

describe('It Should be testing People List Component', () => {
    it('Should test Undefined People list in peoplelist', () => {
         const tree = renderer.create(<PeopleList />).toJSON();
         expect(tree).toMatchSnapshot();
    });

    it('Should test empty Peple List in Component', () => {
        const tree = renderer.create(<PeopleList lstPeoples={[]}/>).toJSON();
        expect(tree).toMatchSnapshot()
    });

    it('Should create the table list of People in Component', () => {
        const tree = renderer.create(<PeopleList lstPeoples={lstPeoples} />).toJSON();
        expect(tree).toMatchSnapshot()
    });

    it('Should render the 1 rows In people List', () => {
        let peopleList = [{
            id: 1,
            username: "Bret",
            email: "Sincere@april.biz",
            phone: "1-770-736-8031 x56442"
        },
        {
            id: 2,
            username: "Lean",
            email: "lean@april.biz",
            phone: "1-770-736-8033 x56442"
        }
        ];

          const tree = renderer.create(<PeopleList lstPeoples={peopleList} />).toJSON();
          expect(tree).toMatchSnapshot()
    });
});