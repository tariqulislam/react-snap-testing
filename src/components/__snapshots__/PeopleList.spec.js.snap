// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It Should be testing People List Component Should create the table list of People in Component 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          User Name
        </th>
        <th>
          Email
        </th>
        <th>
          Phone
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          Bret
        </td>
        <td>
          Sincere@april.biz
        </td>
        <td>
          1-770-736-8031 x56442
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Antonette
        </td>
        <td>
          Shanna@melissa.tv
        </td>
        <td>
          010-692-6593 x09125
        </td>
      </tr>
      <tr>
        <td>
          3
        </td>
        <td>
          Samantha
        </td>
        <td>
          Nathan@yesenia.net
        </td>
        <td>
          1-463-123-4447
        </td>
      </tr>
      <tr>
        <td>
          4
        </td>
        <td>
          Karianne
        </td>
        <td>
          Julianne.OConner@kory.org
        </td>
        <td>
          493-170-9623 x156
        </td>
      </tr>
      <tr>
        <td>
          5
        </td>
        <td>
          Kamren
        </td>
        <td>
          Lucio_Hettinger@annie.ca
        </td>
        <td>
          (254)954-1289
        </td>
      </tr>
      <tr>
        <td>
          6
        </td>
        <td>
          Leopoldo_Corkery
        </td>
        <td>
          Karley_Dach@jasper.info
        </td>
        <td>
          1-477-935-8478 x6430
        </td>
      </tr>
      <tr>
        <td>
          7
        </td>
        <td>
          Elwyn.Skiles
        </td>
        <td>
          Telly.Hoeger@billy.biz
        </td>
        <td>
          210.067.6132
        </td>
      </tr>
      <tr>
        <td>
          8
        </td>
        <td>
          Maxime_Nienow
        </td>
        <td>
          Sherwood@rosamond.me
        </td>
        <td>
          586.493.6943 x140
        </td>
      </tr>
      <tr>
        <td>
          9
        </td>
        <td>
          Delphine
        </td>
        <td>
          Chaim_McDermott@dana.io
        </td>
        <td>
          (775)976-6794 x41206
        </td>
      </tr>
      <tr>
        <td>
          10
        </td>
        <td>
          Moriah.Stanton
        </td>
        <td>
          Rey.Padberg@karina.biz
        </td>
        <td>
          024-648-3804
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`It Should be testing People List Component Should render the 1 rows In people List 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          User Name
        </th>
        <th>
          Email
        </th>
        <th>
          Phone
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          Bret
        </td>
        <td>
          Sincere@april.biz
        </td>
        <td>
          1-770-736-8031 x56442
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          Lean
        </td>
        <td>
          lean@april.biz
        </td>
        <td>
          1-770-736-8033 x56442
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`It Should be testing People List Component Should test Undefined People list in peoplelist 1`] = `
<span>
   Invalide Data Format for People List
</span>
`;

exports[`It Should be testing People List Component Should test empty Peple List in Component 1`] = `
<span>
   Empty List View 
</span>
`;
